premailer/tests/test_cache.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import, unicode_literals
+
+import os
+import unittest
+import imp
+
+import cachetools
+
+
+class TestFunctionCache(unittest.TestCase):
+    def tearDown(self):
+        for key in (
+            "PREMAILER_CACHE",
+            "PREMAILER_CACHE_MAXSIZE",
+            "PREMAILER_CACHE_TTL",
+        ):
+            try:
+                del os.environ[key]
+            except KeyError:
+                pass
+
+    def test_unknown_cache(self):
+        os.environ["PREMAILER_CACHE"] = "UNKNOWN"
+
+        with self.assertRaises(Exception) as assert_context:
+            cache_fd = open("premailer/cache.py", "r")
+            imp.load_source("cache.py", "premailer/cache.py", cache_fd)
+
+        self.assertTrue(
+            assert_context.exception.args[0].startswith(
+                "Unsupported cache implementation"
+            )
+        )
+
+    def test_default_cache(self):
+        cache_fd = open("premailer/cache.py", "r")
+        cache_module = imp.load_source("cache.py", "premailer/cache.py", cache_fd)
```

Comment:
I'm just uncomfortable that this won't be left open for the duration of the whole test suite. Or will `cache_fd.close` always be called at the end of the test method?

Reply:
> Why is the file called `.red`?
This was added by mistake. Removed now.